common/workunit/workunit.cpp

```diff
@@ -6260,6 +6281,15 @@ void CLocalWorkUnit::noteFileRead(IDistributedFile *file)
     }
 }
 
+void CLocalWorkUnit::noteFieldUsage(IPropertyTree * fieldUsage)
+{
+    if (fieldUsage)
+    {
+        CriticalBlock block(crit);
+        p->addPropTree("usedsources", LINK(fieldUsage));
```

Comment:
Sounds good. At some point when this is functional we will run it with Valgrind to find any leaks we have introduced